app/src/main/java/emu/skyline/EmulationActivity.kt

```diff
@@ -477,6 +477,16 @@ class EmulationActivity : AppCompatActivity(), SurfaceHolder.Callback, View.OnTo
             val x = 0f.coerceAtLeast(pointer.x * 1280 / view.width).toInt()
             val y = 0f.coerceAtLeast(pointer.y * 720 / view.height).toInt()
 
+            var attribute = 0
```

Comment:
Do the if statement as if-else and use Kotlin inline if assignment syntax: `val attribute = if (action == ...) 1 else if (...) 2 else 0` and maybe even skip assigning the variable and directly assign it to the array.

Reply:
I don't think is a good idea to avoid assigning the variable since it looses the meaning on the array.

Reply:
As Android Studio will warn you, more than two conditions shouldn't be using an inline if else, but instead be using [when](https://www.baeldung.com/kotlin/when#1-when-as-an-expression)

Reply:
Much better looking alternative. Thank you for the suggestion.